logins.js

```diff
@@ -9,107 +9,111 @@ var oauth_service = 'audience:server:client_id:848232511240-7so421jotr2609rmqakc
 var app = 'com.nianticlabs.pokemongo';
 var client_sig = '321187995bc7cdc2b5fc91b11a96e2baa8602c62';
 
-module.exports = {
-    PokemonClub: function (user, pass, self, callback) {
-        var options = {
+
+var PokemonClub = function (user, pass, self, callback) {
+    var options = {
+        url: login_url,
+        headers: {
+            'User-Agent': 'niantic'
+        }
+    };
+
+    self.request.get(options, function (err, response, body) {
+        if (err) {
+            return callback(err, null);
+        }
+
+        var data = null;
+        try {
+          data = JSON.parse(body);
+        }catch(err){
+          return callback(err, null);
+        }
+
+        options = {
             url: login_url,
+            form: {
+                'lt': data.lt,
+                'execution': data.execution,
+                '_eventId': 'submit',
+                'username': user,
+                'password': pass
+            },
             headers: {
                 'User-Agent': 'niantic'
             }
         };
 
-        self.request.get(options, function (err, response, body) {
-            var data;
-
-            if (err) {
+        self.request.post(options, function (err, response, body) {
+            //Parse body if any exists, callback with errors if any.
+            if(err) {
                 return callback(err, null);
             }
-            
-            try {
-              data = JSON.parse(body);
-            }catch(err){
-              return callback(err, null);
+
+            if (body) {
+                var parsedBody = JSON.parse(body);
```

Comment:
All good :+1:


On Fri, Jul 22, 2016 at 7:57 AM, Joseba Legarreta <notifications@github.com>
wrote:

> In logins.js
> <https://github.com/Armax/Pokemon-GO-node-api/pull/86#discussion_r71892359>
> :
>
> >                  return callback(err, null);
> >              }
> > -
> > -            try {
> > -              data = JSON.parse(body);
> > -            }catch(err){
> > -              return callback(err, null);
> > +
> > +            if (body) {
> > +                var parsedBody = JSON.parse(body);
>
> Ups! My fault you're right 👍
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Armax/Pokemon-GO-node-api/pull/86/files/86e235adeaa7510833de87a7cf8392c1ac4ed5e3..01559e8a53f2ce17d305129c91982e4d55180bac#r71892359>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACzIMJvhOmLSbohtDSxRXDLdpitUPp31ks5qYNpAgaJpZM4JS1CB>
> .
>



-- 
Mushaheed S. Kapadia
Cell: (201)-970-0786